client/src/utils/tone/index.ts

```diff
@@ -0,0 +1,87 @@
+import store from 'store';
+import { FlashMessageArg } from '../../components/Flash/redux';
+import { Themes } from '../../components/settings/theme';
+
+// TODO: Add more tone types
+export async function playTone(
+  state: FlashMessageArg['message'] | Themes | string
+): Promise<void> {
+  const playSound = !!store.get('fcc-sound');
+  if (playSound) {
+    const tone = await import('tone');
+    if (tone.context.state !== 'running') {
+      void tone.context.resume();
+    }
+    let player;
+    switch (state) {
+      case 'flash.incomplete-steps':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/try-again.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case 'flash.cert-claim-success':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/cert.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case Themes.Night:
+        const nightToDayPlayer = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/day.mp3'
+        ).toDestination();
+        if (!nightToDayPlayer.loaded)
+          await nightToDayPlayer.load(
+            'https://campfire-mode.freecodecamp.org/day.mp3'
+          );
+        nightToDayPlayer.start();
+        break;
+      case Themes.Default:
+        const dayToNightPlayer = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/night.mp3'
+        ).toDestination();
+        if (!dayToNightPlayer.loaded)
+          await dayToNightPlayer.load(
+            'https://campfire-mode.freecodecamp.org/night.mp3'
+          );
+        dayToNightPlayer.start();
+        break;
+      case 'donation':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/donate.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case 'tests-complete':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/chal-comp.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case 'completion':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/chal-comp.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case 'tests-failed':
+        player = new tone.Player(
+          'https://campfire-mode.freecodecamp.org/try-again.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      case 'block-toggle':
+        player = new tone.Player(
+          'https://tonejs.github.io/audio/berklee/guitar_chord1.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
+      default:
+        player = new tone.Player(
+          'https://tonejs.github.io/audio/berklee/guitar_chord1.mp3'
+        ).toDestination();
+        player.autostart = playSound;
+        break;
```

Comment:
Do we need a default case here?

It runs this sound on every flash message, because the flash component calls the playsound.

Reply:
Do we not want that?

Reply:
NVM, I see we did not originally have that.

Reply:
At the same time, it feels like something is missing, when I change other settings, and the flash pops-up, but there is no sound

Reply:
That would be neat, sure - however, the implementation would need tweaking.

Currently, if I toggle the theme, the `playSound` gets called by the theme saga and plays the correct sound, and then gets called by the flash component with `flash.updated-settings` which triggers the default case. I end up with two different sounds playing at once and it is cacophonous.

Reply:
> I end up with two different sounds playing at once and it is cacophonous.

So... I do not see the problem - only a feature

Reply:
It sounds like jQuery.